src/session/session_compact.c

```diff
@@ -371,10 +371,12 @@ __compact_worker(WT_SESSION_IMPL *session)
         for (i = 0; i < session->op_handle_next; ++i)
             WT_WITH_DHANDLE(session, session->op_handle[i], __wt_tree_modify_set(session));
         WT_ERR(__compact_checkpoint(session));
+        WT_STAT_CONN_SET(session, session_table_compact_passes, loop);
```

Comment:
Should we set this at the start of the loop? If something goes wrong within that for loop, we want to know compaction was happening. I am wondering if we should keep the counter if something goes wrong instead of setting it back to 0.

Reply:
Adding it at the start does make sense. I'm not sure about keeping the counter set though. It would be confusing for someone unfamiliar with the statistic to observe `session_table_compact_running=0` while the counter is > 0. A counter that reaches 99 will still be visible in the statistics timeline.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9995/commits/f384933c389aa392a0a3f25a74a7a0bcc3766809